### ISSUE TITLE
Certificate verification with trusted private CA fails

### ISSUE THREAD

**Issue by philpennock** [redsolution/xabber-android#285]
Xabber rejects as untrustworthy the certificate chain from my server.

My server uses a certificate from my own CA; as well as DANE trust anchors in DNS, which I don't expect to be usable with Android, I have my CA as a "Trusted credentials"/"User" such that KitKat keeps complaining about the presence of the CA. So I'm highly confident that it's present in the system trust chain.  Android 4.4.2 on Nexus 4, build KOT49H.

Using a browser to visit web-sites with certs from my CA works.  If Xabber could either trust the CA, or at least detect it and say "Hey, we don't know the CA this is using, but the CA is on your system; it's FooBarBaz; want to trust it for this account?" then this would be more usable.

Thanks for your hard work on Xabber. :)

**Comment by andrewnenakhov** [comment]
Just accept connection after warning on bad certificate, or install a letsencrypt cert! )

**Comment by philpennock** [comment]
* Let's Encrypt does not support `xmppAddr` address types.
* Let's Encrypt does not support `SRVName` address types.
* (Let's Encrypt didn't exist when this bug was filed)

**Comment by andrewnenakhov** [comment]
@philpennock actually we do run Xabber.com and xabber.org with letsencrypt and we do use srv records.

As for issue, I believe it is safe to leave it closed.

**Comment by philpennock** [comment]
Leave closed, fine, your project. I'm replying to note that I did not say SRV records, as found in DNS.  I referred to the `SRVName` address-type.  That's as found in the X509v3 subjectAltNames extension, specified in RFC 4985 (and obviously is tied into DNS SRV records).

That the Xabber client doesn't need or check that address-type in a cert's SAN is independent of the address-type's utility for other clients and my wish to keep using my own CA.

Let's Encrypt is great!  It works well for many use-cases, I've donated money to their running, I love it.  Let's Encrypt is not a panacea which lets applications ignore any CAs but those they bundle internally.
